Barotrauma/BarotraumaShared/SharedSource/Events/Missions/AbandonedOutpostMission.cs

```diff
@@ -315,12 +315,17 @@ protected override void UpdateMissionSpecific(float deltaTime)
                 endTimer += deltaTime;
                 if (endTimer > EndDelay)
                 {
-#if SERVER
-                    if (!(GameMain.GameSession.GameMode is CampaignMode) && GameMain.Server != null)
+                    if (GameMain.GameSession.GameMode is not CampaignMode)
                     {
-                        GameMain.Server.EndGame();                        
-                    }
+#if SERVER
+                        GameMain.Server?.EndGame();
+#else
+                        if (GameMain.GameSession.GameMode is { IsSinglePlayer: true } && GameMain.GameSession.CrewManager is { IsSinglePlayer: true })
```

Comment:
I'm not sure I understand why `IsSinglePlayer: true` is checked on both the GameMode and the CrewManager. Is there a situation in which one is singleplayer, and the other isn't?

Reply:
Fixed, it now only checks if the GameMode has IsSinglePlayer set. Also fixed in CoOpMode.cs.